lib/iris/tests/unit/experimental/ugrid/test_MeshCoord.py

```diff
@@ -431,5 +425,274 @@ def test_basic(self):
+        co_facey = co_facex.copy()
+        co_facey.rename("latitude")
+        co_facey.long_name = "face_y"
+        co_edgey = co_edgex.copy()
+        co_edgey.rename("edge_y")
+        co_edgey.long_name = "edge_y"
+
+        face_node_conn = Connectivity(
+            inds_start_index
+            + (
+                face_nodes_array.transpose()
+                if inds_src_dim == 1
+                else face_nodes_array
+            ),
+            cf_role="face_node_connectivity",
+            long_name="face_nodes",
+            start_index=inds_start_index,
+            src_dim=inds_src_dim,
+        )
+
+        edge_node_conn = Connectivity(
+            inds_start_index
+            + (
+                edge_nodes_array.transpose()
+                if inds_src_dim == 1
+                else edge_nodes_array
+            ),
+            cf_role="edge_node_connectivity",
+            long_name="edge_nodes",
+            start_index=inds_start_index,
+            src_dim=inds_src_dim,
+        )
+
+        self.mesh = Mesh(
+            topology_dimension=2,
+            node_coords_and_axes=[(co_nodex, "x"), (co_nodey, "y")],
+            connectivities=[face_node_conn, edge_node_conn],
+            face_coords_and_axes=[(co_facex, "x"), (co_facey, "y")],
+            edge_coords_and_axes=[(co_edgex, "x"), (co_edgey, "y")],
+        )
+
+        # Construct a test meshcoord.
+        meshcoord = MeshCoord(mesh=self.mesh, location=location, axis="x")
+        self.meshcoord = meshcoord
+        return meshcoord
+
+    def _check_expected_points_values(self):
+        # The points are just the face_x-s
+        meshcoord = self.meshcoord
+        self.assertArrayAllClose(meshcoord.points, self.face_xs)
+
+    def _check_expected_bounds_values(self, facenodes_changes=None):
+        mesh_coord = self.meshcoord
+        # The bounds are selected node_x-s :  all == node_number + 100.0
+        result = mesh_coord.bounds
+        # N.B. result should be masked where the masked indices are.
+        expected = 100.0 + self.face_nodes_array
+        if facenodes_changes:
+            # ALSO include any "bad" values in that calculation.
+            bad_values = (self.face_nodes_array < 0) | (
+                self.face_nodes_array >= self.n_nodes
+            )
+            expected[bad_values] = np.ma.masked
+        # Check there are *some* masked points.
+        n_missing_expected = 1
+        if facenodes_changes:
+            n_missing_expected += facenodes_changes["n_extra_bad_points"]
+        self.assertEqual(np.count_nonzero(expected.mask), n_missing_expected)
+        # Check results match, *including* location of masked points.
+        self.assertMaskedArrayAlmostEqual(result, expected)
+
+    def test_points_values(self):
+        """Basic points content check, on real data."""
+        meshcoord = self.meshcoord
+        self.assertFalse(meshcoord.has_lazy_points())
+        self.assertFalse(meshcoord.has_lazy_points())
```

Comment:
@pp-mo Shouldn't this be `has_lazy_bounds()`?

Reply:
:+1:

Reply:
Ok, will fix.